frontend/src/components/MainComponent/MainComponent.tsx

```diff
@@ -167,6 +180,9 @@ function MainComponent({
 			: setMessages(chatHistory);
 
 		setDefences(defences);
+
+		// we will only update the chatModel if it is defined in the backend response. It will only defined for the sandbox level.
+		chatModel && setChatModel(chatModel);
```

Comment:
Is this correct? It means if we switch from sandbox to any other level, we will still have sandbox chatModel in state. Not only is that an unnecessary use of memory, but it's also an inconsistency.

Reply:
I had reasons for this but actually they're bogus. Changed